src/app/config/user/userHooks.js

```diff
@@ -1,21 +1,55 @@
 import user from "../../utilities/api-clients/user";
 import { useEffect, useState } from "react";
+import { startRefeshAndSession } from "../../config/user/userActions";
+import sessionManagement from "../../utilities/sessionManagement";
+import { getAuthState } from "../../utilities/auth";
+import fp from "lodash/fp";
 
-export const useGetPermissions = userToken => {
+export const useGetPermissions = (authState, setShouldUpdateAccessToken) => {
     const [userState, setUserState] = useState();
     useEffect(() => {
-        if (!userToken) {
+        if (!authState) {
             user.getPermissions().then(userData => {
                 // TODO this needs to be removed when we get a correct 401 status back from zebedee.
                 if (userData === "Access Token required but none provided.") {
                     console.error("Error fetching permissions");
                     return null;
                 }
+                setShouldUpdateAccessToken(true);
                 setUserState(userData);
             });
         } else {
-            setUserState(userToken);
+            setShouldUpdateAccessToken(false);
+            setUserState(authState);
         }
     }, []);
     return userState;
 };
+
```

Comment:
This hook makes it very easy to remove the timers from the Layout component should there be any issues